app/com/linkedin/drelephant/tuning/hbt/FitnessManagerHBT.java

```diff
@@ -128,6 +133,48 @@ protected void computeFitness(JobSuggestedParamSet jobSuggestedParamSet, JobExec
     }
   }
 
+  /**
+   * It AutoTuining fault in retry , then apply Penalty.
+   * If Its not autotuning fault , then check , how many times
+   * this parameter applied to the retried execution . If its greater
+   * than threshold , then apply penalty else this parameter is eligible
+   * for rebirth
+   * @param jobSuggestedParamSet
+   * @param jobExecution
+   */
+  private void handleRetriedCases(JobSuggestedParamSet jobSuggestedParamSet, JobExecution jobExecution) {
+    if (jobExecution.autoTuningFault) {
+      logger.info(" AutoTuning fault and hence applying penalty to the parameter ");
+      applyPenalty(jobSuggestedParamSet, jobExecution);
+    } else {
+      logger.info(" Retry is not because of AutoTuning  ");
+      List<TuningJobExecutionParamSet> tuningJobExecutionParamSets = TuningJobExecutionParamSet.find.select("*")
```

Comment:
It would be good to check if it is with in last few runs, rather than looking into all runs.